### ISSUE TITLE
Login error messages not getting fully propagated to the UI

### ISSUE THREAD

**Issue by shana** [github/VisualStudio#351]
When a user hasn't verified their email, octokit returns

```
{"message":"Validation Failed","errors":[{"resource":"OauthAccess","code":"custom","field":"user","message":"user must have a verified email address in order to authenticate via OAuth"}],"documentation_url":"https://developer.github.com/v3/oauth_authorizations/#get-or-create-an-authorization-for-a-specific-app"}    at Octokit.Connection.HandleErrors(IResponse response) in Y:\jenkins\workspace\GitHub for Visual Studio speakeasy jenkinskeep\submodules\octokit.net\Octokit\Http\Connection.cs:line 563
```

but our login UI just says `"Could not connect to github.com"`. We need to propagate these messages properly so users don't get confused about why things are failing.

**Comment by grokys** [comment]
:+1: Yes, this bit me just the other day.

**Comment by grokys** [comment]
Fixed by #352
